variables.tf

```diff
@@ -276,12 +276,7 @@ variable "object_lock_configuration" {
 
 variable "website_inputs" {
 
-  type = list(object({
-    index_document           = string
-    error_document           = string
-    redirect_all_requests_to = string
-    routing_rules            = string
-  }))
+  type    = map(any)
   default = null
```

Comment:
could we keep the variable type the same for backwards compatibility?

Reply:
Sure, but as linked 'redirect_all_requests_to' is mutually exclusive with the other 3 options.. So, you are either setting 'redirect_all_requests_to' or 'index_document', 'error_document' and 'routing_rules'. Not sure how to handle that? I saw there is an experimental 'optional' feature but that is subject to change.

Reply:
Hi, I would really appreciate some feedback to this, not sure how to handle it.

Reply:
the workaround has been to set unused options to null but maybe your way is the correct way forward, provided backwards compatibility is maintained

Reply:
Since this never worked in v2.0, I'm OK with having a breaking change here. Unfortunately, the "optional" feature is not usable (changes to it have been announced).

Let's make this 2 inputs. 
1. `website_redirect_all_requests_to` which is a `list(object)` with `host_name` and `protocol` and a validation of `length <= 2` and default `[]`
2. `website_configuration` which is `list(object)` with the remaining configuration, including a list of `routing_rule` objects,  and a validation of `length <= 2` and default `[]`

What do you think @jurgen-weber-deltatre  and @nitrocode ?